moped-editor/src/components/GridTable/Search.js

```diff
@@ -236,34 +124,6 @@ const Search = ({
     setAdvancedSearchAnchor(null);
   };
 
-  /**
-   * Update the export whenever limit or selectall change
-   */
-  useEffect(
-    () => {
-      if (dialogOpen && !downloading) {
-        query.limit = 0;
-        getExport();
-        setDownloading(true);
-      }
-
-      if (dialogOpen && downloading && data && !loading) {
-        const formattedData = formatExportData(data[query.table]);
-        // use the papaparse library to "unparse" a json object to csv
-        // escapeFormulae: "field values that begin with =, +, -, @, \t, or \r, will be prepended with a ' to defend
-        // against injection attacks because Excel and LibreOffice will automatically parse such cells as formulae."
-        const csvString = Papa.unparse(formattedData, { escapeFormulae: true });
-        setTimeout(() => {
-          // Update the state
-          setDialogOpen(false);
-          downloadFile(csvString);
-        }, 1500);
-      }
-    },
-    // eslint-disable-next-line
-    [dialogOpen, downloading, loading, query.limit, getExport]
-  );
```

Comment:
This useEffect was a workaround for an issue with Apollo Client running the `onCompleted` hook on every render. I demo'd using `onCompleted` today before I realized that it would trigger a CSV download every time the table order or filters changed. Luckily, `getExport` the function that `useLazyQuery` returns to trigger the query fetch now returns a promise that exposes the data object. You can see that further down in `handleExportButtonClick`.

Reply:
@mddilley i'm seeing some odd behavior that sounds similar to what you're describing here.

1. Type `stassney` in the search bar and press `enter` to search
2. Click the download button to download—download works as expected
3. Use the backspace key to clear the search box
4. You are prompted for download again 🙃

Reply:
i'm going to take a  look at this today - thanks for the heads up! 🙏

Reply:
nevermind! i didn't have your most recent commits locally 🤦